feat: add command to run opened chk file

arifszn opened the pull request:
Close #30.
Close #31.

The command can be run from the command palette or by keyboard shortcut: `alt+ctrl+r`. Should the keybinding be changed?

Also, not tested on Mac.

Commit:
refactor: move commands to separated folders

Commit:
feat: add command to run opened .chk file

Commit:
feat: determine command type conditionally while running file

Commit:
feat: add key binding for `chkware.runFile` command

Commit:
refactor: add example comment in `getCommandType`

0hsn commented on package.json:
```diff
@@ -39,6 +40,18 @@
+    "keybindings": [
+      {
+        "command": "chkware.runFile",
+        "key": "ctrl+i ctrl+i",
```
```suggestion
        "key": "alt+ctrl+r",
```

0hsn commented on package.json:
```diff
@@ -39,6 +40,18 @@
+      {
+        "command": "chkware.runFile",
+        "key": "ctrl+i ctrl+i",
+        "mac": "cmd+i cmd+i"
```
```suggestion
        "mac": "opt+ctrl+r"
```

arifszn replied:
Should it be `opt+cmd+r`?

0hsn commented on src/commands/run-file.ts:
```diff
@@ -0,0 +1,65 @@
+    vscode.window.activeTextEditor;
+
+  if (!activeTextEditor) {
+    vscode.window.showWarningMessage("Please open a chk file first.");
```
```suggestion
    vscode.window.showWarningMessage("Please open a .chk file first.");
```

Does it support markdown? then:

```suggestion
    vscode.window.showWarningMessage("Please open a `.chk` file first.");
```

0hsn replied:
would it be possible to enable this key or dropdown item from command-pallet only when a .chk file is open in editor section? .. just asking

arifszn replied:
Yes possible. Added in https://github.com/chkware/vscode-ext/pull/42/commits/39b3156ee2e1699e22f28e3852596ae5653f8fa1.

0hsn commented on src/commands/run-file.ts:
```diff
@@ -0,0 +1,65 @@
+
+  // check file extension and command type
+  if (!openedFileName.endsWith(".chk") || !commandType) {
+    vscode.window.showWarningMessage("Not a valid chk file to run.");
```
```suggestion
    vscode.window.showWarningMessage("Not a valid .chk file to run.");
```

Commit:
chore: change keybinding for `chkware.runFile`

Commit:
refactor: change error messages

Commit:
chore: remove implicit activation events

It became unnecessary to put commands into the activationEvents with vscode v1.74 Stable.

Commit:
feat: enable `chkware.runFile` command for .chk file only

arifszn commented:
Changed have been made according to the reviews.

0hsn reviewed (commented):
I couldn't run it. Therefore need your help for local test on MacOS. 🙏

arifszn commented:
> I couldn't run it. Therefore need your help for local test on MacOS. 🙏

What problem are you facing? Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?

0hsn commented:
> > I couldn't run it. Therefore need your help for local test on MacOS. 🙏
> 
> What problem are you facing? Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?


> What problem are you facing?

- When if press `f5` the extension gets loaded in debug mode.
- I then press `cmd + shift + p` to open command pallet - which works okay, then
    1) I open a new .chk snippet, DO NOT save it, and hit `ctrl + i, ctrl + i` but it do not run
    2) I open a new .chk snippet, THEN save it, and hit `ctrl + i, ctrl + i` but it do not run

> Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?

Yes did so!

Thanks

arifszn commented:
> > > I couldn't run it. Therefore need your help for local test on MacOS. 🙏
> > 
> > 
> > What problem are you facing? Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?
> 
> > What problem are you facing?
> 
> * When if press `f5` the extension gets loaded in debug mode.
> * I then press `cmd + shift + p` to open command pallet - which works okay, then
>   
>   1. I open a new .chk snippet, DO NOT save it, and hit `ctrl + i, ctrl + i` but it do not run
>   2. I open a new .chk snippet, THEN save it, and hit `ctrl + i, ctrl + i` but it do not run
> 
> > Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?
> 
> Yes did so!
> 
> Thanks

The shortcut key has changed in favor of https://github.com/chkware/vscode-ext/pull/42#discussion_r1070513157. PR description has been updated.

0hsn commented:
> > > > I couldn't run it. Therefore need your help for local test on MacOS. 🙏
> > > 
> > > 
> > > What problem are you facing? Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?
> > 
> > 
> > > What problem are you facing?
> > 
> > 
> > 
> > * When if press `f5` the extension gets loaded in debug mode.
> > * I then press `cmd + shift + p` to open command pallet - which works okay, then
> >   
> >   1. I open a new .chk snippet, DO NOT save it, and hit `ctrl + i, ctrl + i` but it do not run
> >   2. I open a new .chk snippet, THEN save it, and hit `ctrl + i, ctrl + i` but it do not run
> > 
> > > Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?
> > 
> > 
> > Yes did so!
> > Thanks
> 
> The shortcut key has changed in favor of [#42 (comment)](https://github.com/chkware/vscode-ext/pull/42#discussion_r1070513157). PR description has been updated.

Unfortunatly shortcut do not works for me. Then I further investigated and found, no shortcut was registered at all for chkware. So, how do the extension register this automatically?

FWIW: I tried `npm install`, then `npm run watch`, then `f5` then `cmd+shift+p`, and search for run. This time the command runs. but, not with shortcut.

![image](https://user-images.githubusercontent.com/3197377/216822014-977764ed-c66d-4914-a2e5-c026d02dc944.png)

Also, in case of when I tried with other file extension, the run command goes away!

arifszn committed:
fix: remove faulty keybinding for mac

arifszn commented:
- > Unfortunatly shortcut do not works for me.

Can you please try now?

- > Also, in case of when I tried with other file extension, the run command goes away!

Shouldn't it be like this as discussed on https://github.com/chkware/vscode-ext/pull/42#discussion_r1070513589?

0hsn commented:
Awesome. This works for me now! Good work. 💟 

> Shouldn't it be like this as discussed on https://github.com/chkware/vscode-ext/pull/42#discussion_r1070513589?

I thought the option is going to stay, in disabled view, rather vanished at all. However, it's okay as well for now. We'll check this capability later. 🙏
